charm.go

```diff
@@ -99,3 +101,39 @@ func IsUnsupportedSeriesError(err error) bool {
 	_, ok := err.(*unsupportedSeriesError)
 	return ok
 }
+
+// MaybeCreateVersionFile creates/overwrite charm version file.
+func MaybeCreateVersionFile(path string) error {
+	var cmdArgs []string
+	// Verify that it is revision control directory.
+	if _, err := os.Stat(filepath.Join(path, ".hg")); err == nil {
+		cmdArgs = []string{"hg", "id", "-n"}
+	} else if _, err = os.Stat(filepath.Join(path, ".bzr")); err == nil {
+		cmdArgs = []string{"bzr", "version-info"}
+	} else if _, err = os.Stat(filepath.Join(path, ".git")); err == nil {
+		cmdArgs = []string{"git", "describe", "--dirty"}
+	} else {
+		logger.Debugf("Charm is not in revision control directory")
+		return nil
+	}
+
+	outStr, err := utils.RunCommand(cmdArgs[0], cmdArgs[1:]...)
+	if err != nil {
+		return err
+	} else {
+		versionPath := filepath.Join(path, "version")
+		// Overwrite the existing version file.
+		file, err := os.OpenFile(versionPath, os.O_RDWR|os.O_CREATE, 0644)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+
+		_, err = file.WriteString(outStr)
```

Comment:
We generally inline things like this as:

if _, err = file.WriteString(outStr); err != nil {
    return err
}

Reply:
done